tests/e2e_tests/astrom_e2e.py

```diff
@@ -0,0 +1,185 @@
+import argparse
+import os
+import pytest
+import numpy as np
+import astropy.time as time
+import astropy.io.fits as fits
+import astropy.io.ascii as ascii
+import corgidrp
+import corgidrp.data as data
+import corgidrp.mocks as mocks
+import corgidrp.walker as walker
+import corgidrp.caldb as caldb
+import corgidrp.detector as detector
+
+thisfile_dir = os.path.dirname(__file__) # this file's folder
+
+def test_astrom_e2e(tvacdata_path, e2eoutput_path):
+    # figure out paths, assuming everything is located in the same relative location
+    l1_datadir = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "L1")
+    processed_cal_path = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "Cals")
+    noise_characterization_path = os.path.join(tvacdata_path, "TV-20_EXCAM_noise_characterization", "darkmap")
+
+    # make output directory if needed
+    astrom_cal_outputdir = os.path.join(e2eoutput_path, "astrom_cal_output")
+    if not os.path.exists(astrom_cal_outputdir):
+        os.mkdir(astrom_cal_outputdir)
+
+    # assume all cals are in the same directory
+    nonlin_path = os.path.join(processed_cal_path, "nonlin_table_240322.txt")
+    dark_path = os.path.join(processed_cal_path, "dark_current_20240322.fits")  
+    flat_path = os.path.join(processed_cal_path, "flat.fits")
+    fpn_path = os.path.join(processed_cal_path, "fpn_20240322.fits")
+    cic_path = os.path.join(processed_cal_path, "cic_20240322.fits")
+    bp_path = os.path.join(processed_cal_path, "bad_pix.fits")
+
+    # create raw data that includes injected stars with gaussian psfs
+    jwst_calfield_path = os.path.join(os.path.dirname(thisfile_dir), "test_data", "JWST_CALFIELD2020.csv")
+    jwst_calfield = ascii.read(jwst_calfield_path)
+    v_mags = jwst_calfield['VMAG']
+    amplitudes = np.power(10, ((v_mags - 22.5) / (-2.5))) * 10
+    image_sources = mocks.create_astrom_data(jwst_calfield_path, add_gauss_noise=False)
+    rows, cols, r0c0 = detector.unpack_geom('SCI', 'image')
+    # create a directory in the output dir to hold the simulated data files
+    rawdata_dir = os.path.join(astrom_cal_outputdir, 'data')
+    if not os.path.exists(rawdata_dir):
+        os.mkdir(rawdata_dir)
+
+    for dark in os.listdir(noise_characterization_path)[:5]:  ## only using the first five frames to save comp time
+        with fits.open(os.path.join(noise_characterization_path, dark)) as hdulist:
+            dark_dat = hdulist[1].data
+            hdulist[0].header['OBSTYPE'] = "AST"
+            rms_noise = np.sqrt(np.mean(dark_dat.flatten())**2)
```

Comment:
I'm not sure this is the rms_noise in the darks (note that taking the sqrt of something that was just squared just cancels out). Probably the best way to get the rms noise is to look at the standard deviation of the dark pixels over time (across multiple files).